### PR TITLE
Romanesque Culture Work

### PR DESCRIPTION
**Pull request by LT-Rascek**
Mechanics:
- Minorities:
   - Faith Converison (by event/decision/interface): a large minority of your old faith will still exist in your capital if you convert
   - Culture creation removes parent culture minorities in counties of the new culture
   - If a hybrid or diverged culture is within the cooldown period, the parent cultures are subsumed instead of becoming a minority when the county culture changes
   - Faith conversion (by any means) now will keep a large minority of the old faith in your capital county

Defines & Game Rules:
- Added default game rule slowing hybrid and divergence
- Reduced a number of hard-coded defines, slowing the growth of cultural acceptance

Cultures:
- Add ability to have more than one parent culture for historical cultures (enabling hybridization locks)
- Updated custom culture name triggers so they should appear more often in games
- Sardinian starts with Proto-Sardinian instead of Sardinian Vulgar in 476

Traditions:
- Roman/Latin Traditions have feature reducing Dissolution Factions for Roman Empires
- Extended Byzantine Dissolution Faction blocker to all Roman Empires
- Frankish Traditions have looser can_keep conditions to model the continuation of Salic Law into the Medieval Period

Traditions:
- Added parameter to Colonizers so that your heritage and language are kept during event hybridization and the capital county's ethos is adopted
- Added parameter to Culture Blending & Malleable Invaders, where the capital county's heritage and language are kept during hybridization and your ethos is adopted

Events:
- Overhauled Vanilla Culture Hybridization Event
   - By default, the hybrid culture uses the capital county's heritage and language, along with your ethos
   - If you have Colonizers tradition, instead your heritage and language are used by the hybrid, along with the capital county culture's ethos
   - If you have Malleable Invaders or Culture Blending, the default is always used
   - Extended Staunch Traditionalist hybrid event protection to a number of other cultures (Greek, Egyptian, Armenian, &c.)
- "Intransigent Marital Practices" event
   - Fires if:
    - A culture head converts to a reformed faith from an unreformed faith
    - He controls 50%+1 of the culture counties
    - The unreformed faith is present in 50%+1 of the culture counties
   - The culture gains the corresponding marriage type tradition of the old faith

Bug Fixes:
- Fixed a bug with the Fall of Rome Event
- Fixed a bug with some combinations of religious reformation and tenets

Localization:
- Renamed "Byzantine Traditions" to "Eastern Roman Traditions"
- Added localization for case where Roman culture heritage drifts in the heritage/language drift events

### CODE REVIEW

File: ChangeLog.txt [hunk 1]
```diff
@@ -1,6 +1,61 @@
...
+Defines & Game Rules:
+- Added default game rule slowing hybrid and divergence
+- Reduced a number of hard-coded defines, slowing the growth of cultural acceptance
+
+Cultures:
+- Add ability to have more than one parent culture for historical cultures (enabling hybridization locks)
+- Updated custom culture name triggers so they should appear more often in games
+- Culture Tweaks:
+   - Sardinian starts with Proto-Sardinian instead of Sardinian Vulgar in 476
+   - Subei (as in, Iberian Suebi) replaces Agrarian with Strong Believers
```
Comment (loup99) [inline ChangeLog.txt:20]: Typo?
```suggestion
   - Suebi (as in, Iberian Suebi) replaces Agrarian with Strong Believers
```

File: ChangeLog.txt [hunk 2]
```diff
@@ -1,6 +1,61 @@
...
+   - Lusitanian is a parent culture of Romano-Hispanic (to prevent hybridization)
+- Tradition Changes:
+   - Visigothic replaces Ruling Caste with Isolationist
+   - Ostrogothic replaces Formation Fighting Experts with Isolationist
+   - Moseogothic replaces Tribal Unity with Isolationist
```
Comment (loup99) [inline ChangeLog.txt:25]: I'm not entirely convinced by these tradition changes. I agree with reducing hybridisation, but as we discussed I think it is a general issue as opposed to specific for these three cultures

File: ChangeLog.txt [hunk 3]
```diff
@@ -1,6 +1,61 @@
...
+   - Skirian replaces Ruling Caste with Loyal Subjects
+   - Indo-Scythian replaces Ruling Caste with Culture Blending/Xenophilic
+   - Alano-Vandalic replaces Ruling Caste with Steppe Tolerance
+
+Traditions:
+- Roman/Latin Traditions have feature reducing Dissolution Factions for Roman Empires
```
Comment (loup99) [inline ChangeLog.txt:31]:
```suggestion
- Roman/Latin Traditions have a feature reducing the number of Dissolution Factions for Roman Empires
```

File: WtWSMS/common/culture/cultures/BP_central_germanic.txt [hunk 4]
```diff
@@ -52,7 +52,7 @@ buri = { # Migrational Era Buri
 		#Ritual Traditions
 		tradition_concubines #Needed for Germanic christians to have polygomy
 		#Realm Traditions
-		tradition_mountain_homes #Monterrei is all mountains
+		tradition_mountain_homes #Monterrei is all mountains and were nestled in teh Carpathians ca. AD Ist Century
```
Comment (loup99) [inline WtWSMS/common/culture/cultures/BP_central_germanic.txt:55]:
```suggestion
		tradition_mountain_homes #Monterrei is all mountains and were nestled in the Carpathians ca. 1st Century AD 
```

File: WtWSMS/common/culture/cultures/BP_east_germanic.txt [hunk 5]
```diff
@@ -293,10 +293,11 @@ visigothic = {
 		#Ritual Traditions
 		tradition_concubines #Source: [1]
 		#Realm Traditions
-		tradition_ruling_caste
+		tradition_isolationist
```
Comment (loup99) [inline WtWSMS/common/culture/cultures/BP_east_germanic.txt:296]: See above

File: WtWSMS/common/on_action/BP_heritage_language_shift_on_actions.txt [hunk 6]
```diff
@@ -47,6 +48,7 @@ liguistic_shift = {
 					HERITAGE_OR_LANGUAGE = language
 					SHIFT_START_YEAR = 500
 					SHIFT_PERIOD_YEARS = 25
+					UNITED_POLITIY_PREVENTS_SHIFT = yes
```
Comment (loup99) [inline WtWSMS/common/on_action/BP_heritage_language_shift_on_actions.txt:51]:
```suggestion
					UNITED_POLITY_PREVENTS_SHIFT = yes
```

File: ChangeLog.txt (hunk 3 above)
Comment (LT-Rascek) [reply ChangeLog.txt:31]: Partly correct; there's only one Dissolution Faction though (which was my typo). Fixed.

File: ChangeLog.txt (hunk 2 above)
Comment (LT-Rascek) [reply ChangeLog.txt]: Then I'll revert these changes; they were made with the idea that Ruling Caste has the same effect on the hybridization event, but that would _also_ make it unlikely for the Visigoths to become a Romance Heritage culture.

File: ChangeLog.txt [hunk 7]
```diff
@@ -40,12 +45,59 @@ Mechanics:
...
+Cultures:
+- Add ability to have more than one parent culture for historical cultures (enabling hybridization locks)
+- Updated custom culture name triggers so they should appear more often in games
+- Culture Tweaks:
+   - Sardinian starts with Proto-Sardinian instead of Sardinian Vulgar in 476
+   - Suebi (as in, Iberian Suebi) replaces Agrarian with Strong Believers
+   - Lusitanian is a parent culture of Romano-Hispanic (to prevent hybridization)
+
+Traditions:
+- Roman/Latin Traditions have a feature reducing occurance of the Dissolution Faction for Roman Empires
```
Comment (loup99) [inline ChangeLog.txt:67]:
```suggestion
- Roman/Latin Traditions have a feature reducing occurrence of the Dissolution Faction for Roman Empires
```

File: WtWSMS/common/scripted_effects/BP_culture_shift_effects.txt [hunk 8]
```diff
@@ -12,16 +12,90 @@ generic_shift_culture_event_effect = { #Cf. Vanilla common/on_actions/yearly_on_
...
-		modifier = { add = 1 current_year >= shift_start_year_1_period }  #02%/year = 39.65% per 25 years
-		modifier = { add = 1 current_year >= shift_start_year_2_period }  #03%/year = 53.30% per 25 years
-		modifier = { add = 2 current_year >= shift_start_year_3_period }  #05%/year = 72.27% per 25 years
-		modifier = { add = 3 current_year >= shift_start_year_4_period }  #08%/year = 87.56% per 25 years
-		modifier = { add = 5 current_year >= shift_start_year_5_period }  #13%/year = 96.92% per 25 years
-		modifier = { add = 8 current_year >= shift_start_year_6_period }  #21%/year = 99.72% per 25 years
-		modifier = { add = 13 current_year >= shift_start_year_7_period } #34%/year = 99.99+% per 25 years
+		#
+		# For the non-mathematically inclined, I've included a few charts for various periods to
+		# give a rough estimate of occurance for choices of period. Approximate year when 50%/90%/99%
```
Comment (loup99) [inline WtWSMS/common/scripted_effects/BP_culture_shift_effects.txt:17]:
```suggestion
		# give a rough estimate of occurrence for choices of period. Approximate year when 50%/90%/99%
```

File: WtWSMS/localization/english/culture/traditions/BP_cultural_languages_l_english.yml [hunk 9]
```diff
@@ -39,7 +38,7 @@
  language_gallic_romance_name: "Gallic Romance"
  language_gandhari_name: "Gandhari"
  language_gaulish_name: "Gaulish"
- language_germano_african_romance_name: "Arréyanu"
+ language_germano_african_romance_name: "Germano-Arréyanu"
```
Comment (loup99) [inline WtWSMS/localization/english/culture/traditions/BP_cultural_languages_l_english.yml:41]: ?

File: WtWSMS/localization/english/event_localization/BP_culture_shift_events_l_english.yml [hunk 10]
```diff
@@ -18,6 +18,7 @@
  ## Language Tooltips
  ## Heritage Drift
  bp_heritage_linguistic_drift.0102.t: "Customary Drift"
+ bp_heritage_linguistic_drift.0102.roman.desc: "Where once all peoples of the Western Mediterranean were united into a single polity with similar behaviours, the fracturing of the Western Empire into dozens, hundreds of states and statelets have effectively halted trade and exchange, sharpening regional differences.\n\nNow the mannerisms and customs of even the [shift_culture.GetName] people has changed enough from the practices of the people who colonized and dominate the #italic Pars Occidentalis#! that Octavian himself would not recognize them."
```
Comment (loup99) [inline WtWSMS/localization/english/event_localization/BP_culture_shift_events_l_english.yml:21]:
```suggestion
 bp_heritage_linguistic_drift.0102.roman.desc: "Where once all peoples of the Western Mediterranean were united into a single polity with similar behaviours, the fracturing of the Western Roman Empire into dozens, hundreds of states and statelets have effectively halted trade and exchange, sharpening regional differences.\n\nNow the mannerisms and customs of even the [shift_culture.GetName] people has changed enough from the practices of the people who colonized and dominate the #italic Pars Occidentalis#! that Octavian himself would not recognize them."
```

File: WtWSMS/localization/english/replace/culture/traditions/cultural_traditions_l_english.yml [hunk 11]
```diff
@@ -1,5 +1,11 @@
 ﻿l_english:
+ #Ethos
  ethos_egalitarian_name: "Cosmopolitan"
+ #Traditions
+ tradition_byzantine_succession_name: "Eastern Roman Traditions"
+ tradition_byzantine_succession_desc: "This culture upholds the legal traditiosn of the Eastern Roman Empire; such laws deeply influence its pratices."
```
Comment (loup99) [inline WtWSMS/localization/english/replace/culture/traditions/cultural_traditions_l_english.yml:6]:
```suggestion
 tradition_byzantine_succession_desc: "This culture upholds the legal traditions of the Eastern Roman Empire; such laws deeply influence its practices."
```

File: ChangeLog.txt [hunk 12]
```diff
@@ -40,12 +45,59 @@ Mechanics:
    - Either of the Senates may collapse if no senator is available to maintain the Senate's functions
    - Rajamandala and Eranshahr governments cannot request interact with the Senate, as they function under very different understanding of universality than the Roman Empire
 
+Previous Mechanics:
+- Minorities:
+   - Faith Converison (by event/decision/interface): a large minority of your old faith will still exist in your capital if you convert
```
Comment (loup99) [inline ChangeLog.txt:50]:
```suggestion
   - Faith Conversion (by event/decision/interface): a large minority of your old faith will still exist in your capital if you convert
```

File: WtWSMS/localization/english/culture/traditions/BP_cultural_languages_l_english.yml (hunk 9 above)
Comment (LT-Rascek) [reply WtWSMS/localization/english/culture/traditions/BP_cultural_languages_l_english.yml:41]: Last I checked we wanted a variant of African Romance with Germanic superstrate; we have it for the other major Romance groups with Germanic kingdoms (e.g., Romano-Gallic, Romano-Hispanic, Italian Romance). There's just no good name for it.

Comment (loup99) [reply WtWSMS/localization/english/culture/traditions/BP_cultural_languages_l_english.yml:41]: Ok, I would probably rather go with African (from African Romance)

Comment (LT-Rascek) [reply WtWSMS/localization/english/culture/traditions/BP_cultural_languages_l_english.yml]: Renamed "Germano-African Romance".

**Review by loup99 — APPROVED** [review]

**Pull request merged**